src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -19,9 +18,47 @@ public class PackageDependencyGroupConverter : JsonConverter
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var set = JsonUtility.LoadJson(reader);
+            string fxName = null;
+            List<Packaging.Core.PackageDependency> packages = new List<Packaging.Core.PackageDependency>();
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType.Equals(JsonToken.PropertyName))
+                {
+                    if (reader.Value.Equals(JsonProperties.Dependencies))
+                    {
+                        // Dependencies are stored in an array
+                        while (reader.Read() && reader.TokenType != JsonToken.EndArray)
+                        {
+                            string id = null;
+                            string version = null;
+                            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+                            {
+                                if (reader.TokenType.Equals(JsonToken.PropertyName) && reader.Value.Equals(JsonProperties.PackageId))
+                                {
+                                    reader.Read();
+                                    id = reader.Value.ToString();
+                                }
+                                if (reader.TokenType.Equals(JsonToken.PropertyName) && reader.Value.Equals(JsonProperties.Range))
+                                {
+                                    reader.Read();
+                                    version = reader.Value.ToString();
+                                }
+                            }
 
-            var fxName = set.Value<string>(JsonProperties.TargetFramework);
+                            if (id != null)
+                            {
+                                packages.Add(new Packaging.Core.PackageDependency(id, string.IsNullOrEmpty(version) ? null : VersionRange.Parse(version)));
+                            }
+                        }
+                        reader.Read();
```

Comment:
```
{
  "TargetFramework": "SomeFramework",
  "Dependencies": [
    {
      "PackageId": "SomeId",
      "Range": "SomeRange"
    }
  ]
}
```

If this is the file we're getting, on line 53 we will move the reader to the end of object, and then on line 23 call read again, I think read should be fine, and return false. 

However, if the part we're reading here is part of a larger file though it will continue reading right? 
```
{ 
  "PackageDependencyGroup": {
    "TargetFramework": "SomeFramework",
    "Dependencies": [
      {
        "PackageId": "SomeId",
        "Range": "SomeRange"
      }
    ]
  },
  "SomethingElse": {
    "uhoh": []
  }
}
```

Reply:
Yeah this also fixed the failing tests